### PR TITLE
[bitnami/discourse] Add feature discourse.plugins

### PR DESCRIPTION
**Pull request by migruiz4**
### Description of the change

Adds a new feature 'discourse.plugins' to the bitnami/discourse chart.

This feature allows users to install plugins during the container initialization, by running `rake plugin:install --repo=...` before the discourse initialization begins.

### Applicable issues

- supersedes #16009

### Additional information

Tested locally using the following values:

```
discourse:
  plugins:
    - https://github.com/discourse/discourse-oauth2-basic
```

The Oauth plugin showing in the /admin/plugins page:

image

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/discourse/templates/deployment.yaml [hunk 1]
```diff
@@ -99,11 +99,26 @@ spec:
...
           args: {{- include "common.tplvalues.render" (dict "value" .Values.discourse.args "context" $) | nindent 12 }}
+          {{- else }}
+          args:
+            - -c
+            - |
+              {{- if .Values.discourse.plugins }}
+              pushd "/opt/bitnami/discourse" >/dev/null || exit
+              {{- range $plugin := .Values.discourse.plugins }}
+              RAILS_ENV=production bundle exec rake plugin:install repo={{ $plugin }}
+              {{- end }}
```
Comment (marcosbc) [inline bitnami/discourse/templates/deployment.yaml:118]: Can you confirm that this works if you upgrade the chart, after having installed some plugins during the 1st initialization?

File: bitnami/discourse/templates/deployment.yaml [hunk 2]
```diff
@@ -99,11 +99,26 @@ spec:
...
           {{- if .Values.diagnosticMode.enabled }}
           args: {{- include "common.tplvalues.render" (dict "value" .Values.diagnosticMode.args "context" $) | nindent 12 }}
           {{- else if .Values.discourse.args }}
           args: {{- include "common.tplvalues.render" (dict "value" .Values.discourse.args "context" $) | nindent 12 }}
+          {{- else }}
+          args:
+            - -c
+            - |
+              {{- if .Values.discourse.plugins }}
+              pushd "/opt/bitnami/discourse" >/dev/null || exit
```
Comment (marcosbc) [inline bitnami/discourse/templates/deployment.yaml:115]:
```suggestion
              pushd "/opt/bitnami/discourse" >/dev/null || exit 1
```

File: bitnami/discourse/templates/deployment.yaml [hunk 3]
```diff
@@ -99,11 +99,26 @@ spec:
...
+              {{- range $plugin := .Values.discourse.plugins }}
+              RAILS_ENV=production bundle exec rake plugin:install repo={{ $plugin }}
+              {{- end }}
+              popd >/dev/null || exit
```
Comment (marcosbc) [inline bitnami/discourse/templates/deployment.yaml:119]:
```suggestion
              popd >/dev/null || exit 1
```

**Review by marcosbc — CHANGES_REQUESTED** [review]
Added a few comments, please have a look at them whenever you have a chance.

File: bitnami/discourse/templates/deployment.yaml (hunk 1 above)
Comment (migruiz4) [reply bitnami/discourse/templates/deployment.yaml:118]: Yes! It works when upgrading.

Plugins will be persisted at `/bitnami/discourse/plugins`, so even if the `discourse.plugins` value is only provided once, they will be persisted between upgrades.

I have tested the following scenarios:

- Install the `bitnami/discourse` chart with Oauth plugin, enable the plugin in the UI, upgrade the chart without providing the plugins value. The chart is upgraded successfully, the plugin and its settings are persisted.
- Install the `bitnami/discourse` chart with Oauth plugin, enable the plugin in the UI, upgrade the chart with the plugins value again. The chart is upgraded successfully, the plugin and its settings are also persisted.

**Review by marcosbc — APPROVED** [review]
LGTM!

**Pull request merged**